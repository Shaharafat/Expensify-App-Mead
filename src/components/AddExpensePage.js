import React from "react";
import { connect } from "react-redux";
import { startAddExpenses } from "../actions/expenses";
import ExpenseForm from "./ExpenseForm";

export class AddExpensePage extends React.Component {
  constructor(props) {
    super(props);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(expense) {
    this.props.startAddExpenses(expense);
    this.props.history.push("/");
  }
  
  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddExpenses: (expenses) => dispatch(startAddExpenses(expenses)),
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
