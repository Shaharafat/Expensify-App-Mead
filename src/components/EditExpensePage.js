import React from "react";
import { connect } from "react-redux";
import { editExpense, removeExpense } from "../actions/expenses";
import ExpenseForm from "./ExpenseForm";

export class EditExpensePage extends React.Component {
  constructor(props) {
    super(props);
    this.editExpense = this.editExpense.bind(this);
    this.removeExpense = this.removeExpense.bind(this);
  }

  editExpense(expense) {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push("/");
  }

  removeExpense() {
    this.props.removeExpense(this.props.expense);
    this.props.history.push("/");
  }

  render() {
    return (
      <div>
        <ExpenseForm expense={this.props.expense} onSubmit={this.editExpense} />
        <button onClick={this.removeExpense}>Remove</button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(
      (expense) => expense.id === props.match.params.id
    ),
  };
};

const mapDispatchToProps = (dispatch) => ({
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  removeExpense: (expense) => dispatch(removeExpense(expense)),
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
