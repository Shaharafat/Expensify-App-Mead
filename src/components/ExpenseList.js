import React from 'react'
import {connect} from 'react-redux'
import ExpenseListItem from './ExpenseListItem'
import selectExpenses from '../selectors/expenses'
const ExpenseList = (props) => (
  <div>
    Expense List
    {
      props.expenses.map(expense => (
        <ExpenseListItem key={expense.id} {...expense} />
      ))
    }
  </div>
)

const mapStateToProps = (state) => {
  return {
    // expenses: state.expenses,
    // filters: state.filters
    expenses: selectExpenses(state.expenses,state.filters),
  }
}

export default connect(mapStateToProps)(ExpenseList)
