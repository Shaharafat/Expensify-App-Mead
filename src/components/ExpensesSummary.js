import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import getVisibleExpenses from "../selectors/expenses";
import getTotalValues from "../selectors/expense-total";

export const ExpensesSummary = ({ length, total }) => {
  const expenseWord = length > 1 ? "expenses" : "expense";
  const formattedTotal = numeral(total / 100).format("$0,0.00");
  return (
    <div>
      <h1>
        {length} {expenseWord} totaling of {formattedTotal}
      </h1>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);
  return {
    length: visibleExpenses.length,
    total: getTotalValues(visibleExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
