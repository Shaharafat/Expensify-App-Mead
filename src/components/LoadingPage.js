import React from 'react'

const LoadingPage = () => {
  return ( 
    <div className="loader">
    <img className="loader-image" src="/images/loader.gif" alt="loading image"/>
    </div>
   );
}
 
export default LoadingPage;