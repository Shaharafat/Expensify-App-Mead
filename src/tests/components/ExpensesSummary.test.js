import React from "react";
import { shallow } from "enzyme";

import {ExpensesSummary} from "../../components/ExpensesSummary";

test("should correctly render ExpenseSummary with 1 expenses", () => {
  const wrapper = shallow(<ExpensesSummary length={1} total={235} />);
  expect(wrapper).toMatchSnapshot();
});

test("should correctly render ExpenseSummary with multiple expenses", () => {
  const wrapper = shallow(<ExpensesSummary length={23} total={235324234} />);
  expect(wrapper).toMatchSnapshot();
});
