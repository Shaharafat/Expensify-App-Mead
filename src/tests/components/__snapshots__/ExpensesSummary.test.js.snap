// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with 1 expenses 1`] = `
<div>
  <h1>
    1
     
    expense
     totaling of 
    $2.35
  </h1>
</div>
`;

exports[`should correctly render ExpenseSummary with multiple expenses 1`] = `
<div>
  <h1>
    23
     
    expenses
     totaling of 
    $2,353,242.34
  </h1>
</div>
`;
